extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java

```diff
@@ -807,12 +809,34 @@ private MSQErrorReport getErrorReportOrThrow(String controllerTaskId)
 
   private void assertMSQSpec(MSQSpec expectedMSQSpec, MSQSpec querySpecForTask)
   {
-    Assert.assertEquals(expectedMSQSpec.getQuery(), querySpecForTask.getQuery());
+    assertMSQSpecQuery(expectedMSQSpec.getQuery(), querySpecForTask.getQuery());
     Assert.assertEquals(expectedMSQSpec.getAssignmentStrategy(), querySpecForTask.getAssignmentStrategy());
     Assert.assertEquals(expectedMSQSpec.getColumnMappings(), querySpecForTask.getColumnMappings());
     Assert.assertEquals(expectedMSQSpec.getDestination(), querySpecForTask.getDestination());
   }
 
+  private void assertMSQSpecQuery(Query msqSpecQuery, Query taskSpecQuery)
+  {
+    Assert.assertEquals(msqSpecQuery.getId(), taskSpecQuery.getId());
```

Comment:
Why do we need to verify each field separately?
Does MSQSpec.Query.equals() fail due to different context values?

Reply:
> Does MSQSpec.Query.equals() fail due to different context values?

Yeah

Reply:
That doesn't seem right. Why are there different context values?

Reply:
@kfaraz Actually no, you're right. This seems an older code change which was needed when we were putting things in the queryContext. This shouldn't be needed anymore, I've reverted this change and am running the MSQ tests locally to double check.

Thanks for pointing this out!

Reply:
Thanks!